meta-lmp-bsp/dynamic-layers/freescale-layer/recipes-bsp/imx-atf/imx-atf/0002-plat-imx8m-add-SiP-call-for-SRC-PERSIST_SECONDARY_BO.patch

```diff
@@ -0,0 +1,113 @@
+From b58afa3bfe3053039a97671d9fc8f8ef200bc86f Mon Sep 17 00:00:00 2001
+From: Igor Opaniuk <igor.opaniuk@foundries.io>
+Date: Wed, 10 Mar 2021 11:50:03 +0200
+Subject: [PATCH] plat: imx8m: add SiP call for SRC PERSIST_SECONDARY_BOOT
+
+This introduces SiP call for retrieving/setting SRC GPR10
+PERSIST_SECONDARY_BOOT bit.
+
+SRC GPR10 PERSIST_SECONDARY_BOOT bit identifies which image is used for
+boot: primary or secondary. As the bit is retained after _warm_ reset,
+that permits to command BootROM to boot needed image after _warm_ reset.
+
+This functionality is handy for the setups with OTA support, where
+are A/B sets of boot firmware (one for regular boot, another for
+recovery).
+
+If the PERSIST_SECONDARY_BOOT is 0, the boot ROM uses address 0x8400 for the
+primary image. If the PERSIST_SECONDARY_BOOT is 1, the boot ROM reads that
+secondary image table from address 0x8200 on the boot media and uses the
+address specified in the table for the secondary image.
+
+Secondary Image Table contains the sector of secondary bootloader image,
+exluding the offset to that image (explained below in the note).
+To generate the Secondary Image Table, use e.g.:
+$ printf '\x0\x0\x0\x0\x0\x0\x0\x0\x33\x22\x11\x00\x00\x01\x0\x0\x00\x0\x0\x0'
+  > /tmp/sit.bin
```

Comment:
the imx implementation pads this file to 512 bytes

Reply:
yes, correct. Will fix commit message